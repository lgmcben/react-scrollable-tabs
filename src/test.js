import React from 'react';
import { shallow } from 'enzyme';
import ScrollableTabs, { Tab } from './'


describe('ScrollableTabs', () => {

  let wrapper;

  beforeEach(
    () => wrapper = shallow(<ScrollableTabs />)
  );

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('execute the passed click handler', () => {
    const mockCallBack = jest.fn();
    wrapper.setProps({ clickHandler: mockCallBack });
    wrapper.simulate('click');
    expect(mockCallBack).toHaveBeenCalled();
  });

})
