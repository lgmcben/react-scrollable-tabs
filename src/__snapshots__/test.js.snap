// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollableTabs renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollableTabs />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  display: flex; // 1
  flex-wrap: nowrap; // 2
  overflow-x: auto; // 3
  -webkit-overflow-scrolling: touch; // 4
  -ms-overflow-style: -ms-autohiding-scrollbar; // 5

  &::-webkit-scrollbar {
    display: none; // 6
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  display: flex; // 1
  flex-wrap: nowrap; // 2
  overflow-x: auto; // 3
  -webkit-overflow-scrolling: touch; // 4
  -ms-overflow-style: -ms-autohiding-scrollbar; // 5

  &::-webkit-scrollbar {
    display: none; // 6
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
